concourse/tasks/terraform-plan.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+set -eu
+
+export TF_IN_AUTOMATION=1
+
+# Assume role once here, rather than configuring the same thing (via different
+# options) in each Terraform provider in every root module.
+#
+# TODO: If this ends up being used elsewhere, factor it out into a script and
+# build it into the image.
+creds="$(aws sts assume-role \
+  --role-session-name "concourse-$(date +%d-%m-%y_%H-%M-%S)" \
+  --role-arn "${ASSUME_ROLE_ARN}" \
+  | jq .Credentials)"
+export AWS_ACCESS_KEY_ID="$(echo $creds | jq -r .AccessKeyId)"
+export AWS_SECRET_ACCESS_KEY="$(echo $creds | jq -r .SecretAccessKey)"
+export AWS_SESSION_TOKEN="$(echo $creds | jq -r .SessionToken)"
+
+cd "${DEPLOYMENT_PATH}"
+
+terraform init -input=false -backend-config "${ENVIRONMENT}.backend"
+
+terraform plan -input=false \
```

Comment:
Please don't copy-paste this file. All you need to do make the script reusable is to change `plan` to `$1` and pass `plan`/`apply` as a positional argument from Concourse (and rename the file to `terraform.sh`).

Reply:
yeah, I know, testing a bit. I don't know if we are interested in that feature too.

Reply:
Cool, no worries then. I might just go ahead and refactor the script anyway - I kinda made it unnecessarily specific before.

Reply:
Thanks. I will put it into review once I'm ready, still investigating some k8s auth.